server/src/controller.ts

```diff
@@ -21,12 +34,20 @@ export let registration = async (req: Request, res: Response) => {
             return;
         }
 
-        await User.create({ username: req.body.username, password: req.body.password }, function (err: any, user: any) {
-            if (err) {
-                res.status(501).send("Error!");
+        await User.findOne({username: req.body.username}, (err: any, user: any) => {
+            if (!user) {
+                User.create({ username: req.body.username, password: req.body.password },
+                        function (err: any, user: any) {
+                    if (err) {
+                        res.status(501).send("Error!");

```

Comment:
501 -> 500

Reply:
I was going to ask you to change it to 400 instead, because this may be a validation error. I mean you can set min length for username for example, and the `User.create` will return a error if we try to create a User having a username with length less than that min length. My point is that failed validation for a field is more of a Bad Request than an Internal Server Error.